libraries/rush-lib/src/logic/operations/ShellOperationRunner.ts

```diff
@@ -138,20 +140,57 @@ export class ShellOperationRunner implements IOperationRunner {
   }
 
   private async _executeAsync(context: IOperationRunnerContext): Promise<OperationStatus> {
-    // TERMINAL PIPELINE:
-    //
-    //                             +--> quietModeTransform? --> collatedWriter
-    //                             |
-    // normalizeNewlineTransform --1--> stderrLineTransform --2--> removeColorsTransform --> projectLogWritable
-    //                                                        |
-    //                                                        +--> stdioSummarizer
+    // Only open the *.cache.log file(s) if the cache is enabled.
+    const cacheProjectLogWritable: ProjectLogWritable | undefined = this._buildCacheConfiguration
+      ? new ProjectLogWritable(
+          this._rushProject,
+          context.collatedWriter.terminal,
+          `${this._logFilenameIdentifier}.cache`
+        )
+      : undefined;
+
     const projectLogWritable: ProjectLogWritable = new ProjectLogWritable(
       this._rushProject,
       context.collatedWriter.terminal,
       this._logFilenameIdentifier
     );
 
     try {
+      //#region CACHE LOGGING
+      const cacheDiscardTransform: DiscardStdoutTransform = new DiscardStdoutTransform({
+        destination: context.collatedWriter
+      });
+
+      let cacheCollatedTerminal: CollatedTerminal;
+      const cacheConsoleWritable: TerminalWritable = context.quietMode
+        ? cacheDiscardTransform
+        : context.collatedWriter;
+      if (cacheProjectLogWritable) {
+        const cacheSplitterTransform: SplitterTransform = new SplitterTransform({
+          destinations: [cacheConsoleWritable, cacheProjectLogWritable]
+        });
+        cacheCollatedTerminal = new CollatedTerminal(cacheSplitterTransform);
+      } else {
+        cacheCollatedTerminal = new CollatedTerminal(cacheConsoleWritable);
+      }
+
+      const buildCacheTerminalProvider: CollatedTerminalProvider = new CollatedTerminalProvider(
+        cacheCollatedTerminal,
+        {
+          debugEnabled: context.debugMode
+        }
+      );
+      const buildCacheTerminal: Terminal | undefined = new Terminal(buildCacheTerminalProvider);
```

Comment:
This can't be undefined when it is `const` and assigned the result of a constructor, unless said constructor contains a `return undefined` inside of itself (which would be quite unusual).

Reply:
Oops, didn't fix this one.